### DISCUSSION TITLE
Vercel

### DISCUSSION THREAD

**Discussion by chris-yooo** [neuefische/questions#56] [Web] [answered]
Problem is that the .env file is not uploaded to github and therefore the connection to the database does not work I would like to fix that :)

this is my vercel link: https://capstone-project-gilt.vercel.app/

link to the repo: https://github.com/chris-yooo/capstone-project

**Comment by F-Kirchhoff** [comment, marked as answer]
Hey Chris,
this is intended, since you never want to push your API keys to github. You can set .env variables in the settings of your vercel project.

Update:

Name: expects the name of the variable
Value:  expects the value of the variable


![Screenshot 2022-08-12 at 11 07 31](https://user-images.githubusercontent.com/59687330/184322962-a931cc25-ee69-47e8-9fbf-5ed20f484901.png)

**Reply by F-Kirchhoff** [reply]
Just make sure you use the same name as in your local environment so your program can find it.

**Reply by chris-yooo** [reply]
ok hab ich getestet
![Screenshot (10)](https://user-images.githubusercontent.com/55069641/184340200-14cbd43d-63bb-4826-9f38-01bed68728fd.png)
und mein erster test war ohne MONGODB_URI= im bereich "value"

geht nich :(

**Reply by ahohnsen** [reply]
In das Feld "value" kommt nur das, was hinter dem "=" steht. :)

**Reply by chris-yooo** [reply]
wie gesagt getestet ohne MONGODB_URI=

klappt nich

**Reply by PReichetanz** [reply]
Auch nicht nach der Integration von MongoDB in Vercel, wie es [in diesem Artikel](https://www.infoworld.com/article/3664936/how-to-deploy-with-vercel-and-mongodb-atlas-without-even-trying.html) erklärt wird mithilfe [dieser Seite](https://vercel.com/integrations/mongodbatlas) (die Links hatte ich dir schon als DM geschickt)?

**Reply by pixelass** [reply]
> ⚠️ @chris-yooo Bitte darauf achten dass du deinen MOGODB_URI key hier nicht sichtbar machst. Wenn der Screenshot deinen echten key einthält rate ich dir diesen zu erneuern da sonst jeder in deine Datenbank schreiben kann.

Der Vorgang wie oben beschrieben ist richtig. 
Wenn das nicht geht, bitte nach weiteren Fehlermeldungen schauen. 
Den log zu Vercel findest du im aktuellen Deployment unter "Functions"

![vercel-function-log](https://user-images.githubusercontent.com/1148334/187922447-eaad2b97-9b5f-455e-bceb-5af7052e8797.gif)

**Reply by mpagels** [reply]
Und nach dem Ändern der `Environment Variabels` immer ein `Redeploy` auf Vercel manuell durchführen, damit die neuen Werte auch im Projekt genutzt werden.
